src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -382,22 +382,38 @@ bool ObjectMonitor::enter_for(JavaThread* locking_thread) {
 }
 
 bool ObjectMonitor::try_enter(JavaThread* current) {
-  void* cur = try_set_owner_from(nullptr, current);
-  if (cur == nullptr) {
-    assert(_recursions == 0, "invariant");
-    return true;
-  }
+  if (LockingMode == LM_LIGHTWEIGHT) {
```

Comment:
Could probably just do this for all LockingModes.

Reply:
Yes, you convinced me that reading the owner for legacy mode doesn't need to be done as the return from the CAS because it can change anyway.  If the owner is the stack lock, which the code is testing for, it will never change here so it's ok to read it again.